pkg/kubelet/kubelet.go

```diff
@@ -1834,6 +1834,23 @@ func (kl *Kubelet) GetPods() []*api.Pod {
 	return kl.podManager.GetPods()
 }
 
+// GetRunningPods returns all pods running on kubelet from looking at the
+// container runtime cache. This function converts kubecontainer.Pod to
+// api.Pod, so only the fields that exist in both kubecontainer.Pod and
+// api.Pod are considered meaningful.
+func (kl *Kubelet) GetRunningPods() ([]*api.Pod, error) {
```

Comment:
The kubelet code all looks reasonably ok to me barring my comments, but cc @dchen1107 for a second set of eyes, as she is much more familiar with this code than I am.